python/setup.py

```diff
@@ -26,6 +29,16 @@
     cpu_count = 1
 
 
+def get_version():
+    here = os.path.abspath(os.path.dirname(__file__))
+    version_file_path = os.path.join(here, 'src/py/keyvi/__init__.py')
+    version_file = codecs.open(version_file_path, 'r').read()
+    version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]", version_file, re.M)
```

Comment:
what about splitting the version into parts, major, minor, etc. Then we can construct the version any way we need it without requiring regex logic.

If we then add a parameter to setup.py (e.g. `--release`) it could omit the `dev0` suffix for releases. That way we do not need another commit for releases. `--release` could be automatically added by our travis script if run from a tag, that would combine our already working automatic release packaging with this.

Reply:
@hendrikmuhs agree! 
It will make development more convenient,  but as you have mentioned that will require some logic to be implemented (which needs to work for `sdist` as well), so I think it's better to add that in a separate PR.

Reply:
I looked a bit around in other projects. Scipy is doing it similar to my suggestion, they have a `ISRELEASED` bool in their setup.py. The release process seems to be: create branch from master and then they commit a change on the release branch that sets `ISRELEASED` to true and on master they bump the version. Lbnl they create a tag on the release branch.

I think that makes sense, otherwise you have to: commit a change to remove 'dev0', tag the release, create a commit that bumps the version. Creating a release branch isn't a bad choice either.

Having all that said, it seems quite some overhead for a small project, but looks like we become more serious, so why not.

Reply:
I was checking `numpy` and they also have `ISRELEASED` bool. So that seems pretty common.

On the other hand the solution with `--release` was quite convenient, but seems it will be a bit tricky to get it work with `sdist`.

So I'll implement the version with `ISRELEASED` bool.